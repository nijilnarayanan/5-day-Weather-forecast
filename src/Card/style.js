import styled from 'styled-components'


export const Wrapper = styled.section`
display: flex;
flex-direction: row;
align-items: center;
justify-content: center;
background-color: #e5e5e5;
margin-top: 10px;
border-radius: 4px;
padding: 8px;

@media screen and (min-width: 767px) {
    flex-direction: column;
    margin-right: 16px;
    padding: 16px;
    height: 200px;
    width: 100px;
    margin-top: 0px;
    border-radius: 8px;
    &:last-child {
        margin-right: 0px;
    }
}
`

export const Day = styled.div`
font-weight: 500;
@media screen and (min-width: 767px) {
    font-size :20px;
    margin-bottom: 6px;
}
`
export const DateTime = styled.div`
margin-left: 8px;
@media screen and (min-width: 767px) {
    margin-left: 0px;
}
`
export const Icon = styled.img`

`
export const Temperature = styled.div`
font-weight: 700;
@media screen and (min-width: 767px) {
    margin-bottom: 6px;
    font-size :20px;
}
`
export const Description = styled.div`
margin-left: 8px;
@media screen and (min-width: 767px) {
    margin-left: 0px;
    font-size :20px;
}
`
