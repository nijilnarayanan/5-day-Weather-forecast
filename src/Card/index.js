import React from 'react'
import { Wrapper, Day, DateTime, Icon, Temperature, Description } from './style'
import moment from 'moment'
import PropTypes from 'prop-types'

const Card = ({ weatherData }) => {
	const {
		dt,
		main: { temp },
		weather,
	} = weatherData || {}
	const { description, icon } = weather[0] || {}

	const weekday = dt * 1000 // converting into milliseconds

	const day = moment(weekday).format('ddd')
	const dateAndTime = moment(weekday).format('Do, MMM')
	const degreeCelsius = temp - 273.15 // converting Kelvin to Degree Celsius

	return (
		<Wrapper>
			<Day>{day}</Day>
			<DateTime>{dateAndTime}</DateTime>
			<Icon src={`https://openweathermap.org/img/w/${icon}.png`} />
			<Temperature>{Math.round(degreeCelsius)}Â°C</Temperature>
			<Description>{description}</Description>
		</Wrapper>
	)
}

Card.defaultProp = {
	weatherData: {},
}
Card.propTypes = {
	weatherData: PropTypes.shape({}),
}

export default Card
