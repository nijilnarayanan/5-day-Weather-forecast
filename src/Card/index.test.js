import React from 'react'
import { mount } from 'enzyme'
import Card from './index'

test('Run Test <Card />', () => {
	const component = mount(<Card />)
	expect(component.debug()).toMatchSnapshot()
})
