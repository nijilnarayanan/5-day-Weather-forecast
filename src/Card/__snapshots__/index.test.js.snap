// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run Test <Card /> 1`] = `
"<Card>
  <styled.article>
    <article className=\\"sc-bdfBwQ hikiyW\\">
      <styled.div>
        <div className=\\"sc-gsTCUz jnUaKZ\\">
          Invalid date
        </div>
      </styled.div>
      <styled.div>
        <div className=\\"sc-dlfnbm jbeaVV\\">
          Invalid date
        </div>
      </styled.div>
      <styled.img src=\\"https://openweathermap.org/img/w/undefined.png\\">
        <img src=\\"https://openweathermap.org/img/w/undefined.png\\" className=\\"sc-hKgILt\\" />
      </styled.img>
      <styled.div>
        <div className=\\"sc-eCssSg fajYQx\\">
          NaN
          Â°C
        </div>
      </styled.div>
      <styled.div>
        <div className=\\"sc-jSgupP cuBoir\\" />
      </styled.div>
    </article>
  </styled.article>
</Card>"
`;
